client/src/models/Location.js

```diff
@@ -32,10 +32,28 @@ export default class Location extends Model {
       lat: yup
         .number()
         .nullable()
+        .min(-90, "Latitude must be a number between -90 and +90")
+        .max(90, "Latitude must be a number between -90 and +90")
+        .test("lat", "Please enter latitude", function(lat) {
+          const { lng } = this.parent
+          if (lng || lng === 0) {
+            return !!lat || lat === 0
+          }
+          return true
+        })
         .default(null),
       lng: yup
         .number()
         .nullable()
+        .min(-180, "Longitude must be a number between -180 and +180")
+        .max(180, "Longitude must be a number between -180 and +180")
```

Comment:
Be aware that this ties it to WGS84, so should be changed when e.g. MGRS support gets added.

Reply:
MGRS is not represented with two scalar values, it is represented as a single string value. that string is either 2, 4, 6, 8, 10, 12 or 14 characters length, depending on location precision. (12 characters for 10m precision) So, i was thinking of storing coordinates as lat/lng no matter which format is chosen. We can convert back and forth while displaying and editing. I have made some experiments with this approach and i couldn't see any major drawbacks with this approach. For validating mgrs values, i also added a new field to `Location` model and used that field on form, but remove it from the model before saving.